language/stdlib/modules/association.move

```diff
@@ -0,0 +1,112 @@
+// Implements logic for registering addresses as association addresses, and
+// determining if the sending account is an association account.
+// Errors:
+// 1000 -> INVALID_GENESIS_ADDR
+// 1001 -> INSUFFICIENT_PRIVILEGES
+// 1002 -> NOT_AN_ASSOCIATION_ACCOUNT
+// 1003 -> ACCOUNT_DOES_NOT_HAVE_PRIVILEGE
+// 1004 -> ACCOUNT_DOES_NOT_HAVE_PRIVILEGE_RESOURCE
+address 0x0:
+
+module Association {
+    use 0x0::Transaction;
+
+    // The root account privilege. This is created at genesis and has
+    // special privileges (e.g. removing an account as an association
+    // account). It cannot be removed.
+    resource struct Root { }
+
+    // There are certain association capabilities that are more
+    // privileged than other association operations. This resource with the
+    // type representing that privilege is published under the privileged
+    // account.
+    resource struct PrivilegedCapability<Privilege> { is_certified: bool }
+
+    // A type tag to mark that this account is an association account.
+    // It cannot be used for more specific/privileged operations.
+    struct T { }
+
+    // Initialization is called in genesis. It publishes the root resource
+    // under the root_address() address, marks it as a normal
+    // association account.
+    public fun initialize() {
+        let sender = Transaction::sender();
+        Transaction::assert(sender == root_address(), 1000);
+        move_to_sender(Root{ });
+        move_to_sender(PrivilegedCapability<T>{ is_certified: true });
+    }
+
+    // Publish a specific privilege under the sending account.
+    public fun apply_for_privilege<Privilege>() {
```

Comment:
We use `register` for this elsewhere, but I like `apply` much better...

Reply:
Same... I vote we keep this, and we can update the names for `register` to `apply` elsewhere :)